docs/guides/deployment/docker.md

```diff
@@ -12,129 +12,227 @@ search:
+```
+
+Next, we'll create a `Dockerfile` that we'll use to create a Docker image that will also store the flow code. 
+
+<div class="terminal">
+```bash
+touch Dockerfile
+```
+</div>
+
+We'll add the following content to our `Dockerfile`:
+
+```dockerfile title="Dockerfile"
+# We're using the latest version of Prefect with Python 3.9
 FROM prefecthq/prefect:2-python3.9
+
+# Add our requirements.txt file to the image and install dependencies
 COPY requirements.txt .
 RUN pip install -r requirements.txt --trusted-host pypi.python.org --no-cache-dir
-ADD flows /opt/prefect/flows
+
+# Add our flow code to the image
+COPY flows /opt/prefect/flows
+
+# Run our flow script when the container starts
+CMD ["python", "flows/prefect-docker-guide-flow.py"]
 ```
 
-Finally, we build this image by running: 
+## Building a Docker image
+
+Now that we have a Dockerfile we can build our image by running: 
 
+<div class="terminal">
 ```bash
-docker build -t docker-tutorial-image .
+docker build -t prefect-docker-guide-image .
 ```
+</div>
 
-This will create a Docker image in your Docker repository with your Prefect flow stored in the image.  
+And we can check that our image works by running:
 
-## Prefect.yaml File
+<div class="terminal">
+```bash
+docker run -e PREFECT_API_URL=YOUR_PREFECT_API_URL -e PREFECT_API_KEY=YOUR_API_KEY prefect-docker-guide-image
+```
+</div>
+
+The container should start up and serve the flow within the container!
+
+!!!tip Prefect API URL and API key
+    In the `docker run` command above, we're passing in the Prefect API URL and API key as environment variables.
+
+    If you are using Prefect Cloud, you will need an Prefect API URL and API key for your served flow to be able to communicate with the Prefect API. You can get an API key from the [API Keys](https://docs.prefect.io/2.12.0/cloud/users/api-keys/) section of the user settings in the Prefect UI.
+
+    You can get the Prefect API URL for your current setup by running `prefect config view` in your terminal.
+
+## Deploying to a remote environment
 
-Now that you have created a Docker image that stores your Prefect flow code, you're going to make use of Prefect's deployment recipes. In your terminal, run:
+Now that we have a Docker image with our flow code embedded, we can deploy it to a remote environment! 
+
+For this guide, we'll simulate a remote environment by using Kubernetes locally with Docker Desktop. You can use the [instructions provided by Docker to set up Kubernetes locally.](https://docs.docker.com/desktop/kubernetes/) 
+
+### Creating a Kubernetes deployment manifest
+
+To ensure the process serving our flow is always running, we'll create a [Kubernetes deployment](https://kubernetes.io/docs/concepts/workloads/controllers/deployment/). If our flow's container ever crashes, Kubernetes will automatically restart it, thus ensuring that we won't miss any scheduled runs.
+
+First, we'll create a `deployment.yaml` file in our `prefect-docker-guide` directory:
```

Comment:
may want to consider a different name here :P

Reply:
Oh yeah, `deployment.yaml` mean somethin' diffrent 'round these parts.